src/ir.cpp

```diff
@@ -19837,6 +19860,18 @@ static uint32_t ptr_len_to_size_enum_index(PtrLen ptr_len) {
     zig_unreachable();
 }
 
+static PtrLen size_enum_index_to_ptr_len(uint32_t size_enum_index) {
```

Comment:
Is there some way to set this up such that if that enum ever changes, we'll get a compile error here, or at least something easily debuggable?

Reply:
I think what I would do is define the enum values of `builtin.Size` in some data structure, then generate both the builtin code and an enum that the C++ compiler can use.  Then instead of using hardcoded values of `0`, `1`, `2`, and `3`, you could use something like `BUILTIN_SIZE_ONE` `BUILTIN_SIZE_MANY` `BUILTIN_SIZE_SLICE` and `BUILTIN_SLICE_C`.

But this design would take a bit of work, I would think it would warrant a separate PR.  Unless you have a simpler idea?

Reply:
I have a simpler idea: `static_assert`